src/STEMMUS_SCOPE.m

```diff
@@ -339,12 +339,26 @@
         % update GroundwaterSettings.headBotmLayer and GroundwaterSettings.tempBotm, from MODFLOW through BMI
         GroundwaterSettings.gw_Dep = groundwater.calculateGroundWaterDepth(GroundwaterSettings.topLevel, GroundwaterSettings.headBotmLayer, ModelSettings.Tot_Depth);
 
-        % update Dunnian runoff and ForcingData.Precip_msr
+        % Update Dunnian runoff and ForcingData.Precip_msr
         [ForcingData.R_Dunn, ForcingData.Precip_msr] = groundwater.updateDunnianRunoff(ForcingData.Precip_msr, GroundwaterSettings.gw_Dep);
 
         % Calculate the index of the bottom layer level
         [GroundwaterSettings.indxBotmLayer, GroundwaterSettings.indxBotmLayer_R] = groundwater.calculateIndexBottomLayer(GroundwaterSettings.soilThick, GroundwaterSettings.gw_Dep);
         [depToGWT_strt, indxGWLay_strt] = groundwater.findPhreaticSurface(SoilVariables.hh, KT, GroundwaterSettings.soilThick, GroundwaterSettings.indxBotmLayer_R);
+
+        % Check the position of the groundwater table
+        if any(GroundwaterSettings.gw_Dep > ModelSettings.Tot_Depth) || any(isempty(GroundwaterSettings.indxBotmLayer))
+            warning('Groundwater table is below the end of the soil column. Please enlarge the total soil thickness!');
+            return
```

Comment:
I add this check here to solve this comment (https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/242#issuecomment-2268550576).

The return statement in this check doesn't work if I add it inside the functions ([calculateGroundWaterDepth](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/cfcefddf59416182aca6cf16deb62b2883a78b40/src/%2Bgroundwater/calculateGroundWaterDepth.m#L9-L11) and [calculateIndexBottomLayer](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/fix_soil_temperature_issues/src/%2Bgroundwater/calculateIndexBottomLayer.m)), and the code still works and run to fail. so I move the check to the main code and it works (when the if statement is True, model stops before failure).